command/run.go

```diff
@@ -132,50 +133,52 @@ func (c *RunCommand) Run(args []string) int {
 	}
 
 	// Get Job struct from Jobfile
-	job, err := c.JobGetter.StructJob(args[0])
+	job, err := c.JobGetter.ApiJob(args[0])
 	if err != nil {
 		c.Ui.Error(fmt.Sprintf("Error getting job struct: %s", err))
 		return 1
 	}
 
-	// Initialize any fields that need to be.
-	job.Canonicalize()
-
-	// Check that the job is valid
-	if err := job.Validate(); err != nil {
-		c.Ui.Error(fmt.Sprintf("Error validating job: %v", err))
+	// Get the HTTP client
+	client, err := c.Meta.Client()
+	if err != nil {
+		c.Ui.Error(fmt.Sprintf("Error initializing client: %s", err))
 		return 1
 	}
 
+	// Force the region to be that of the job.
+	if r := *job.Region; r != "" {
+		client.SetRegion(r)
+	}
+
 	// Check if the job is periodic or is a parameterized job
 	periodic := job.IsPeriodic()
 	paramjob := job.IsParameterized()
 
+	// Check that the job is valid
+	if _, err := client.Jobs().Validate(job, nil); err != nil {
+		c.Ui.Error(fmt.Sprintf("Error validating job: %v", err))
+		return 1
+	}
+
 	// Parse the Vault token
 	if vaultToken == "" {
 		// Check the environment variable
 		vaultToken = os.Getenv("VAULT_TOKEN")
 	}
 
 	if vaultToken != "" {
-		job.VaultToken = vaultToken
-	}
-
-	// Convert it to something we can use
-	apiJob, err := convertStructJob(job)
-	if err != nil {
-		c.Ui.Error(fmt.Sprintf("Error converting job: %s", err))
-		return 1
+		job.VaultToken = helper.StringToPtr(vaultToken)
 	}
 
 	// COMPAT 0.4.1 -> 0.5 Remove in 0.6
-	if apiJob.TaskGroups != nil {
+	if job.TaskGroups != nil {
 	OUTSIDE:
-		for _, tg := range apiJob.TaskGroups {
+		for _, tg := range job.TaskGroups {
 			if tg.Tasks != nil {
 				for _, task := range tg.Tasks {
 					if task.Resources != nil {
-						if task.Resources.DiskMB > 0 {
+						if task.Resources.DiskMB != nil {
```

Comment:
In the validation response we should add a list of warnings and then we can add this to the list and push this server side